### PR TITLE
Focused Launch: add in-memory router

### LINKED ISSUE #46863: Focused Launch: add internal router-like logic for showing detailed views of domain picker and plans grid
We discussed different strategies, as they may have implications on how the modal behaves when the user clicks "back"/"forward" history buttons.

Initial implementation will be only "in memory", meaning that we either implement this logic via component state, or via a "memory router" ([example](https://reactrouter.com/web/api/MemoryRouter)) — the browser's history will not be affected.

**Acceptance criteria:**
- Clicking _Go back_ link in the top left should close the detailed view (plans or domains) and show Focused Launch summary view.
- Clicking the "X" button in the top right of the screen should close the whole Focused Launch modal.
- Re-opening the modal by clicking the "Launch" button again will display the summary view regardless of the last active view.

### PR DESCRIPTION
**Pull request by ciampo**
## Changes proposed in this Pull Request

* Used [`MemoryRouter`](https://reactrouter.com/web/api/MemoryRouter) to handle the routing logic inside the focused launch flow
* Moved the content of the modal into the "summary" route
* Added two additional routes for the domain and plan details

Note: This PR does not contain Typography styles for the router links

### Prepare the sandboxed environment:
* Make sure you have an active connection to your sandbox, and that your sandbox is clean and up to date
* In three separate terminal windows, from the root of the project, run:
  - `yarn start`
  - `cd apps/editing-toolkit && yarn dev --sync`
  - `cd apps/wpcom-block-editor && yarn dev --sync`
* Sandbox an _unlaunched_ WordPress site created via the `/start` site creation flow by adding it to your host file

### Test for changes:
* Visit `/page/SANDBOXED_SITE_ID_CREATED_WITH_START.wordpress.com/home?flags=create/focused-launch-flow`
* Click on "Launch" button — Focused Launch modal should appear
* Click on the "View all domains" link — A new placeholder view for the domain details is shown
* Click on "Go back" — the focused launch summary view is shown
* Click on the "View all plans" link — A new placeholder view for the plan details is shown
* Click on "Go back" — the focused launch summary view is shown
* At any point, the "X" button should be visible in top right of the page. Clicking on this button should close the modal. When re-opening the modal, the summary view should always be shown, regardless of the last active view.

For more details about testing instructions, please check this [screen recording](https://d.pr/v/5nBToY)

Fixes #46863

### CODE REVIEW

**Review by tjcafferkey — APPROVED** [review]
LGTM! Works as expected.

File: packages/launch/package.json [hunk 1]
```diff
@@ -38,7 +38,8 @@
 		"@wordpress/core-data": "^2.16.0",
 		"@wordpress/icons": "^2.4.0",
 		"@wordpress/url": "^2.17.0",
-		"classnames": "^2.2.6"
+		"classnames": "^2.2.6",
+		"react-router-dom": "^5.2.0"
```
Comment (razvanpapadopol) [inline packages/launch/package.json:42]: By using the existing package version from `client` I guess we can get rid of the [yarn.lock noise](https://github.com/Automattic/wp-calypso/pull/47062/files#diff-51e4f558fae534656963876761c95b83b6ef5da5103c4adef6768219ed76c2de) in this PR.

```suggestion
		"react-router-dom": "^5.1.2"
```

**Review by razvanpapadopol — COMMENTED** [review]
It's great to see the routing coming along well 🎉   
I've left some notes but feel free to decide what is worth fixing at this time.

File: packages/launch/src/focused-launch/summary/index.tsx [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+	const title = getSite()?.title;
+
+	const site = useSite();
+
+	const [ siteTitle, setSiteTitle ] = React.useState( title );
+	const [ siteDomainName ] = React.useState( site.currentDomainName );
+	const domainSearch = useDomainSearch();
+
+	return (
+		<div className="summary__container">
```
[the remark this answers is not in the document]
Comment (ciampo) [reply packages/launch/src/focused-launch/summary/index.tsx:42]: Agree. Unfortunately the linter forces the classnames to follow the folder name.

Screenshot 2020-11-04 at 14 47 02

Maybe I should change the folder name to `focused-launch-summary`? Or do you have any other suggestion?

File: packages/launch/src/focused-launch/plan-details/index.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { FocusedLaunchRoute } from '../routes';
+
+import './style.scss';
+
+const FocusedLaunchPlanDetails: React.FunctionComponent = () => {
```
[the remark this answers is not in the document]
Comment (ciampo) [reply packages/launch/src/focused-launch/plan-details/index.tsx]: Updated

File: packages/launch/src/focused-launch/summary/index.tsx [hunk 4]
```diff
@@ -0,0 +1,155 @@
+/**
+ * External dependencies
+ */
+import { Title } from '@automattic/onboarding';
+import { useSelect } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { TextControl, SVG, Path } from '@wordpress/components';
+import React from 'react';
+import DomainPicker from '@automattic/domain-picker';
+import { useSite, useDomainSearch } from '../../hooks';
```
[the remark this answers is not in the document]
Comment (ciampo) [reply packages/launch/src/focused-launch/summary/index.tsx]: Done

File: packages/launch/package.json (hunk 1 above)
Comment (ciampo) [reply packages/launch/package.json]: Downgraded to `5.1.2`. This PR doesn't affect anymore `yarn.lock`

**Comment by ciampo** [comment]
All feedback has been addressed, apart from https://github.com/Automattic/wp-calypso/pull/47062#discussion_r517293404

File: packages/launch/src/focused-launch/summary/index.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (razvanpapadopol) [reply packages/launch/src/focused-launch/summary/index.tsx:43]: Using `/* eslint-disable wpcalypso/jsx-classname-namespace */` is common in these cases where you have a sub-component that should follow namespace from parent.

Just a note for the future: we can start using CSS-is-JS in packages following the pattern from `composite-checkout`.

**Review by razvanpapadopol — APPROVED** [review]
Looking good 🚀

File: packages/launch/src/focused-launch/summary/index.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (ciampo) [reply packages/launch/src/focused-launch/summary/index.tsx]: I disable the linter rule and renamed the css classname to start with `focused-launch-summary__` instead

I opened https://github.com/Automattic/wp-calypso/issues/47112 to track the conversion to CSS-in-JS

**Pull request merged**